File: apps/mobile/modules/apple-pay-intent/ios/ApplePayIntentModule.swift
import ExpoModulesCore
import UIKit

public class ApplePayIntentModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ApplePayIntent")

    OnCreate {
      registerKeboShortcuts()
    }

    Function("syncToken") { (accessToken: String, refreshToken: String) in
      TokenManager.shared.setAccessToken(accessToken)
      TokenManager.shared.setRefreshToken(refreshToken)
    }

    Function("clearToken") {
      TokenManager.shared.clearTokens()
    }

    Function("syncApiUrl") { (url: String) in
      TokenManager.shared.setApiUrl(url)
    }

    AsyncFunction("getFailedTransactions") { () -> String in
      let queue = FailedTransactionQueue.shared
      let transactions = queue.getAll()
      guard let data = try? JSONEncoder().encode(transactions) else {
        return "[]"
      }
      return String(data: data, encoding: .utf8) ?? "[]"
    }

    Function("clearFailedTransactions") {
      FailedTransactionQueue.shared.clearAll()
    }

    AsyncFunction("openShortcutsApp") {
      await MainActor.run {
        if let url = URL(string: "shortcuts://app") {
          UIApplication.shared.open(url)
        }
      }
    }
  }
}

File: apps/mobile/modules/apple-pay-intent/src/ApplePayIntentModule.ts
import { requireNativeModule } from "expo-modules-core";

interface ApplePayIntentModuleType {
  syncToken(accessToken: string, refreshToken: string): void;
  clearToken(): void;
  getFailedTransactions(): Promise<string>;
  clearFailedTransactions(): void;
  syncApiUrl(url: string): void;
  openShortcutsApp(): Promise<void>;
}

export default requireNativeModule<ApplePayIntentModuleType>(
  "ApplePayIntent"
);
